### PR TITLE
feat: adding svelte support for Cypress

### PR DESCRIPTION
**Pull request by JessicaSachs**
### Context

I was set to give a demo of component testing with Svelte tomorrow as a preview for the official Alpha release/support in late September. Is there an open issue for Svelte support already? I know it's on the roadmap.

- Closes 

### User facing changelog

Adding Alpha support for Vanilla Svelte (this does not add support for SvelteKit).

### Additional details

The Svelte Testing Library was my reference implementation.

### Steps to test

1. Create a new Svelte project
2. Try to mount and test Svelte components

### How has the user experience changed?

We need to add gifs to the docs.

### PR Tasks

**TODOS:** We need a Svelte testing guide in the docs and we need to test vanilla Svelte against a variety of library versions using System Tests. Additionally, we need to fix the types for the mounting libraries.
- [ ] Have tests been added/updated?
- [ ] Has the original issue (or this PR, if no issue exists) been tagged with a release in ZenHub? (user-facing changes only)
- [ ] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [ ] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?

### CODE REVIEW

File: cli/package.json [hunk 1]
```diff
@@ -141,6 +142,11 @@
       "require": "./react/dist/cypress-react.cjs.js",
       "types": "./react/dist/index.d.ts"
     },
+    "./svelte": {
```
Comment (lmiller1990) [inline cli/package.json:145]: I looked into this - these entries actually do not do anything from what I can tell, or, at least, they are not respected by webpack 4. I am pretty sure we don't need these at all.

What's actually happening is the bundlers look at `node_modules/cli/{react,vue}` and look inside for `package.json`, and resolve from there.

File: npm/svelte/cypress.config.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  responseTimeout: 2500,
+  projectId: '134ej7',
+  experimentalFetchPolyfill: true,
+  e2e: {
+    supportFile: false,
+  },
+  component: {
+    devServer: {
+      bundler: 'vite',
+      framework: 'vue',
```
Comment (lmiller1990) [inline npm/svelte/cypress.config.ts:16]: `svelte`?

**Review by lmiller1990 — COMMENTED** [review]
Thanks for this!! Someone from @cypress-io/component-testing will need to add the handler to webpack/vite, etc.

File: cli/package.json (hunk 1 above)
Comment (JessicaSachs) [reply cli/package.json:145]: They're for types and Vite. If you take them out, rebuild, and try to do a deep import from `cypress/vue`, Vite will say that it cannot find that module.

File: npm/svelte/cypress.config.ts (hunk 2 above)
Comment (JessicaSachs) [reply npm/svelte/cypress.config.ts:16]: Yeah. I missed one.
```suggestion
      framework: 'svelte',
```

**Comment by JessicaSachs** [comment]
Hmm, the `props` aren't getting passed through even though we're matching the API for testing-library. I'm surprised by that.

File: npm/svelte/src/index.ts [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+
+    el.append(componentNode)
+
+    // mount the component using VTU and return the component on Cypress.svelteComponent
+    const ComponentConstructor = Component.default || Component
+
+    const component = Cypress.svelteComponent = new ComponentConstructor({
+      target: componentNode,
+      options,
+    })
```
Comment (PatrickG) [inline npm/svelte/src/index.ts:96]: > Hmm, the `props` aren't getting passed through even though we're matching the API for testing-library. I'm surprised by that.

Saw your talk on SvelteSociety.
There is no `options` property on the [`options` parameter of a svelte component constructor](https://svelte.dev/docs#run-time-client-side-component-api-creating-a-component).

I guess this should be:
```ts
    const component = Cypress.svelteComponent = new ComponentConstructor({
      ...options,
      target: componentNode,
    })
```

Comment (JessicaSachs) [reply npm/svelte/src/index.ts:96]: _Thank you_!

File: npm/svelte/src/index.ts [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+    componentNode.id = '__cy_svelte_root'
+
+    el.append(componentNode)
+
+    // mount the component using VTU and return the component on Cypress.svelteComponent
+    const ComponentConstructor = Component.default || Component
+
+    const component = Cypress.svelteComponent = new ComponentConstructor({
+      target: componentNode,
+      options,
```
Comment (JessicaSachs) [inline npm/svelte/src/index.ts:95]:
```suggestion
      ...options,
```

**Comment by ZachJW34** [comment]
@JessicaSachs I'm picking up Svelte support! Probably going to merge this work into a feature branch at some point

**Pull request merged**